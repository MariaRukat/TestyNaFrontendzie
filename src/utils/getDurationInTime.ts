import {
  PromotionDurationOutput,
  PromotionDurationType,
} from "../types/PromotionDurationType";

export const getDurationInTime = (
  startDate: Date,
  endDate: Date
): PromotionDurationOutput => {
  let timeDiff = Math.abs(endDate.getTime() - startDate.getTime());
  let hh: string | number = Math.floor(timeDiff / 1000 / 60 / 60);
  hh = `0${hh}`.slice(-2);

  timeDiff -= +hh * 1000 * 60 * 60;
  let mm: string | number = Math.floor(timeDiff / 1000 / 60);
  mm = `0${mm}`.slice(-2);

  timeDiff -= +mm * 1000 * 60;
  let ss: string | number = Math.floor(timeDiff / 1000);
  ss = `0${ss}`.slice(-2);

  return {
    type: PromotionDurationType.time,
    value: {
      hours: +hh,
      minutes: +mm,
      seconds: +ss,
    },
  };
};
