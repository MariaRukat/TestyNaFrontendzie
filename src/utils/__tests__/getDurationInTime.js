import { getDurationInTime } from "../getDurationInTime";

describe("getDurationInTime", () => {
  test("should return time object for timestamp", () => {
    const startDate = new Date("2020-02-11 10:00");
    const endDate = new Date("2020-02-11 10:30");

    const result = getDurationInTime(startDate, endDate);
    expect(result.value).toEqual({ hours: 0, minutes: 30, seconds: 0 });
    expect(result.type).toEqual("time");
  });

  it("should return time between two dates with hours, minutes and seconds", () => {
    const startDate = new Date("2020-02-11 10:00:00");
    const endDate = new Date("2020-02-11 13:30:21");

    const result = getDurationInTime(startDate, endDate);
    expect(result.value).toMatchSnapshot();
    expect(result.type).toEqual("time");
  });
});
