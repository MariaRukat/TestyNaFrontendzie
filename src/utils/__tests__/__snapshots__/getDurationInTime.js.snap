// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDurationInTime should return time between two dates with hours, minutes and seconds 1`] = `
Object {
  "hours": 3,
  "minutes": 30,
  "seconds": 21,
}
`;
